### ISSUE TITLE
cb and cr are the wrong way around in the green decoding

### ISSUE THREAD

**Issue by TheRealAaronPrince** [NuclearLighthouseStudios/Cassette-Video-Decoder#3]
I was trying to debug why the blacks were green and the whites were magenta. after tinkering with the values I found swapping the cr and cb in the calculation of green fixed the issue. I am mainly trying to find out if anyone else is having this issue or if i've done something wrong

**Comment by TheRealAaronPrince** [comment]
never mind, I'd done something that had eliminated green entirely

**Comment by Foxy6670** [comment]
I've figured out the cause of the issue (or... at least, I think so.)

I've also been encountering this problem whenever I'm not using a loopback device. It has to do with the way the audio is encoded.

The audio in the WAV file is _stereo_, while - more than most likely - your hardware is _mono_. I was able to recreate this issue by converting the WAV to mono and trying to decode it. The result is a pink and green mess.

Therefore, you'd need to get stereo equipment to make this work. Kind of a bummer...

**Comment by Foxy6670** [comment]
Yep! As it turns out, this was directly mentioned in issue #2:

> OK, I was able to reproduce your issue by mixing down audio of a similar picture to mono. So it looks like Safari only provides a mono input when using `getUserMedia`. Might need some experimentation to see if there is a workaround by specifying the right constraints.

_Originally posted by @NuclearLighthouseStudios in https://github.com/NuclearLighthouseStudios/Cassette-Video-Decoder/issues/2#issuecomment-767611101_

So unless Y+CbCr (both chromas on one carrier) support is implemented (or a new `mono` mode), this won't work on `mono` hardware.

**Comment by TheRealAaronPrince** [comment]
yes I think that was the issue, as I've been using something else (nbtv) and came across a similar issue. the audio inputs on my computer are mono, except for the media in on the back, which hilariously has the issue that it inverts the audio signal.

tldr: mono input, not software issue.
